tests/test_web_sendfile_functional.py

```diff
@@ -238,47 +246,64 @@ async def handler(request):
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
-    resp = await client.get("/")
+    resp = await client.get(
+        "/", headers={"Accept-Encoding": accept_encoding} if accept_encoding else None
+    )
     assert resp.status == 200
     body = await resp.read()
     assert b"hello aiohttp\n" == body
     assert resp.headers["Content-Type"] == "application/pdf"
-    assert resp.headers.get("Content-Encoding") == "gzip"
+    assert resp.headers.get("Content-Encoding") == expect_encoding
     resp.close()
     await resp.release()
     await client.close()
 
 
+@pytest.mark.parametrize(
+    ("accept_encoding", "expect_encoding"),
+    [(None, "br"), ("gzip, deflate", "gzip")],
+)
+@pytest.mark.parametrize("forced_compression", [None, web.ContentCoding.gzip])
 async def test_static_file_with_gziped_counter_part_enable_compression(
-    aiohttp_client: Any, sender: Any
+    aiohttp_client: Any,
+    sender: Any,
+    accept_encoding: Optional[str],
+    expect_encoding: str,
+    forced_compression: Optional[web.ContentCoding],
 ):
-    """Test that enable_compression does not double compress when a .gz file is also present."""
+    """Test that enable_compression does not double compress when a static compressed file is also present."""
     filepath = pathlib.Path(__file__).parent / "hello.txt"
 
     async def handler(request):
         resp = sender(filepath)
-        resp.enable_compression()
+        resp.enable_compression(forced_compression)
         return resp
 
     app = web.Application()
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
-    resp = await client.get("/")
+    resp = await client.get(
+        "/", headers={"Accept-Encoding": accept_encoding} if accept_encoding else None
+    )
```

Comment:
Please avoid branching in tests

Alternative suggestion https://github.com/bdraco/aiohttp/commit/f0fb424d8b151f715243247483e42a7e99d3cd44

Reply:
A test for an empty `Accept-Encoding` header and `*`  as well would be good since https://www.rfc-editor.org/rfc/rfc9110#section-12.5.3 shows them as potential payloads

Reply:
Wanted to say the same. Logic in tests makes them more fragile and harder to follow.

Reply:
> Please avoid branching in tests
> 
> Alternative suggestion [bdraco@f0fb424](https://github.com/bdraco/aiohttp/commit/f0fb424d8b151f715243247483e42a7e99d3cd44)

I understand - I just didn't view it that way since `headers={"Accept-Encoding": None}` fails as it probably should, so `None` is just a signal to remove the header.

Anyway, thinking more about this, I would instead recommend to revert the [commit to test with the default client header](https://github.com/aio-libs/aiohttp/commit/b0d58f7c6164bff5ca59fd3e97cfeb652eac3ef5).  IMHO, your version reduces the readability and detracts from the point of the test, which is to test the **server** response to **known** `Accept-Encoding` headers.  It shouldn't care about the `aiohttp` client default.  For example:

- If the client changed its default to "gzip, deflate", the test would fail even though it shouldn't.
- If the client changed its default to "br, deflate", the test is weakened because the expectation of `.br` over `.gz` is not tested.

If it doesn't already exist somewhere in the suite, there should be a client test that when no headers are specified, assert that the expected default headers are transmitted to the server in the final request.


> A test for an empty `Accept-Encoding` header and `*` as well would be good since https://www.rfc-editor.org/rfc/rfc9110#section-12.5.3 shows them as potential payloads

Completely agree, but I feel it's flirting with the boundary of scope creep here.  We've already established a few issues with the spec compliance of interpreting the `Accept-Encoding` header, so adding such tests should wait for a correct parsing and also alter the tests when `enable_compression` is used, which is purposefully left out here.  I just wrote up https://github.com/aio-libs/aiohttp/issues/8104 to cover this.write up an issue

Reply:
> Anyway, thinking more about this, I would instead recommend to revert the [commit to test with the default client header](https://github.com/aio-libs/aiohttp/commit/b0d58f7c6164bff5ca59fd3e97cfeb652eac3ef5)

I think thats fine. The goal of the suggestion was to preserve what you had without branching in the test.

Reply:
> Completely agree, but I feel it's flirting with the boundary of scope creep here. We've already established a few issues with the spec compliance of interpreting the `Accept-Encoding` header, so adding such tests should wait for a correct parsing and also alter the tests when `enable_compression` is used, which is purposefully left out here. I just wrote up #8104 to cover this.

Its fine to do it in another PR. I'd like to solve it before merging this one to avoid conflicts since we can backport fixes to 3.9 but this PR will only go back to 3.10 since its a new feature.

Reply:
> Its fine to do it in another PR. I'd like to solve it before merging this one to avoid conflicts since we can backport fixes to 3.9 but this PR will only go back to 3.10 since its a new feature.

Resolving the branching issue here.  Let's discuss backporting on #8104.